### PR TITLE
Update QR code rendering to support VerificationRequests

### PR DESCRIPTION
**Pull request by turt2live**
Makes for building the QR code easier and more common.

No visual changes.

**Requires https://github.com/matrix-org/matrix-js-sdk/pull/1190**

### CODE REVIEW

**Comment by turt2live** [comment]
Theoretically makes https://github.com/vector-im/riot-web/issues/11217 magic alongside other work

File: src/components/views/elements/crypto/VerificationQRCode.js [hunk 1]
```diff
@@ -31,13 +34,81 @@ export default class VerificationQRCode extends React.PureComponent {
         // User verification use case only
         secret: PropTypes.string,
         otherUserKey: PropTypes.string, // Base64 key being verified
-        requestEventId: PropTypes.string,
+        otherUserDeviceKey: PropTypes.string, // Base64 key of the other user's device (optional)
```
Comment (dbkr) [inline src/components/views/elements/crypto/VerificationQRCode.js]: might be helpful to clarify here that this is what we *think* their device key is (presumably?)

File: src/components/views/elements/crypto/VerificationQRCode.js [hunk 2]
```diff
@@ -31,13 +34,81 @@ export default class VerificationQRCode extends React.PureComponent {
...
+        const requestEvent = verificationRequest.requestEvent;
+        qrProps.requestEventId = requestEvent.getId()
+            ? requestEvent.getId()
+            : ToDeviceChannel.getTransactionId(requestEvent);
+
+        // Populate the keys we need depending on which direction and users are involved in the verification.
+        if (myUserId === otherUserId) {
+            if (!otherDeviceId) {
+                // New -> Existing session QR code
+                qrProps.otherUserDeviceKey = null;
```
Comment (dbkr) [inline src/components/views/elements/crypto/VerificationQRCode.js]: what are the semantics of the empty string vs null here?

File: src/components/views/elements/crypto/VerificationQRCode.js [hunk 3]
```diff
@@ -31,13 +34,81 @@ export default class VerificationQRCode extends React.PureComponent {
...
+
+        const requestEvent = verificationRequest.requestEvent;
+        qrProps.requestEventId = requestEvent.getId()
+            ? requestEvent.getId()
+            : ToDeviceChannel.getTransactionId(requestEvent);
+
+        // Populate the keys we need depending on which direction and users are involved in the verification.
+        if (myUserId === otherUserId) {
+            if (!otherDeviceId) {
+                // New -> Existing session QR code
```
Comment (dbkr) [inline src/components/views/elements/crypto/VerificationQRCode.js]: might be clearer to phrase this in terms of which device is doing the showing/scanning - I'm not sure which way round this is from the comment

File: src/components/views/elements/crypto/VerificationQRCode.js (hunk 2 above)
Comment (turt2live) [reply src/components/views/elements/crypto/VerificationQRCode.js]: just doesn't show up in the rendered URL iirc

File: src/components/views/elements/crypto/VerificationQRCode.js (hunk 3 above)
Comment (turt2live) [reply src/components/views/elements/crypto/VerificationQRCode.js]: tbh that feels more confusing, but will try.

**Pull request merged**